src/sagemaker/serve/utils/hardware_detector.py

```diff
@@ -48,55 +48,56 @@
 }
 
 
-def _get_gpu_info(instance_type: str, session: Session) -> int:
+def _get_gpu_info(instance_type: str, session: Session) -> tuple:
     """Get GPU info for the provided instance"""
     ec2_client = session.boto_session.client("ec2")
+    ec2_instance = _format_instance_type(instance_type)
+
+    try:
+        instance_info = ec2_client.describe_instance_types(InstanceTypes=[ec2_instance]).get(
+            "InstanceTypes"
+        )[0]
+    except ClientError as e:
+        logger.error("Failed to describe instance types %s", e)
+        raise ValueError("Provided instance_type is not GPU enabled.")
 
-    split_instance = instance_type.split(".")
-    split_instance.pop(0)
-
-    ec2_instance = ".".join(split_instance)
+    if instance_info is not None:
+        gpus_info = instance_info.get("GpuInfo")
+        if gpus_info is not None:
+            instance_gpu_info = (
+                gpus_info.get("Gpus")[0].get("Count"),
```

Comment:
`gpus_info.get("Gpus")[0].get("Count")` and could be none `gpus_info.get("TotalGpuMemoryInMiB")` in some cases (I'm actually not entirely sure, could be helpful if you can run this against a comprehensive list), do we need to handle them

Reply:
it should be less of an issue tho

Reply:
Should not cause issue, as they are expected values.